admin/services/group.service.js

```diff
@@ -1,49 +1,116 @@
 'use strict'
 
 const groupDataService = require('../services/data-access/group.data.service')
-const pupilDataService = require('../services/data-access/pupil.data.service')
-
 const groupService = {}
 
+/**
+ * Get groups.
+ * @param query
+ * @returns {Promise<Promise|*>}
+ */
 groupService.getGroups = async function (query) {
-  return groupDataService.getGroups(query)
+  // @TODO: TO BE DELETED
+  // return groupDataService.getGroups(query)
+  return groupDataService.sqlGetGroups()
 }
 
+/**
+ * Get pupils filtered by schoolId and groupId.
+ * @param schoolId
+ * @param groupIdToExclude
+ * @returns {Promise<*>}
+ */
 groupService.getPupils = async function (schoolId, groupIdToExclude) {
```

Comment:
Why do want to exclude a single group?  Why not make it an array of groups that could be excluded?  Could you supply some more context please?

Reply:
From what i have learnt today, this is to get the pupils that are not in any group so they can be added to one.  However there is too much going on in one function.  needs better name and separation